react-app/src/features/package/package-activity/hook.test.tsx

```diff
@@ -0,0 +1,141 @@
+import { renderHook, waitFor } from "@testing-library/react";
+import { beforeEach, describe, expect, it, Mock, vi } from "vitest";
+import { useAttachmentService } from "./hook";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { getAttachmentUrl } from "@/api";
+import JSZip from "jszip";
+import { saveAs } from "file-saver";
+
+const wrapper = ({ children }) => (
+  <QueryClientProvider client={new QueryClient()}> {children}</QueryClientProvider>
+);
+
+vi.mock("@/api", async (importOriginals) => ({
+  ...(await importOriginals()),
+  getAttachmentUrl: vi.fn(),
+}));
+
+vi.mock("file-saver", async (importOriginals) => ({
+  ...(await importOriginals()),
+  saveAs: vi.fn(),
+}));
+
+global.fetch = vi.fn();
+
+describe("useAttachmentService", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("calls onUrl getAttachmentUrl with correct parameters", async () => {
+    const packageId = "testPackage";
+    const attachment = {
+      title: "Test Bucket",
+      bucket: "testBucket",
+      key: "testKey",
+      filename: "testFile",
+      uploadDate: 123232,
+    };
+
+    const mockedGetAttachmentUrl = getAttachmentUrl as Mock;
+    mockedGetAttachmentUrl.mockResolvedValue("http://example.com/testFile");
+
+    const { result } = renderHook(() => useAttachmentService({ packageId }), {
+      wrapper,
+    });
+
+    const url = await result.current.onUrl(attachment);
+
+    expect(mockedGetAttachmentUrl).toHaveBeenCalledWith(
+      packageId,
+      attachment.bucket,
+      attachment.key,
+      attachment.filename,
+    );
+
+    expect(url).toBe("http://example.com/testFile");
+  });
+
+  it("calls onZip and generates a ZIP and downloads it with correct filenames", async () => {
+    const packageId = "testPackage";
+    const attachments = [
+      {
+        title: "Test Bucket",
+        bucket: "bucket1",
+        key: "key1",
+        filename: "file1.md",
+        uploadDate: 123231,
+      },
+      {
+        title: "Test Bucket",
+        bucket: "bucket2",
+        key: "key2",
+        filename: "file2.md",
+        uploadDate: 123232,
+      },
+    ];
+
+    const mockedGetAttachmentUrl = getAttachmentUrl as Mock<typeof getAttachmentUrl>;
```

Comment:
is there a reason why you put a type here but not line 40? just curious!

Reply:
Nope. Resolving with https://github.com/Enterprise-CMCS/macpro-mako/pull/859/commits/3ddff667bce06303808b83d4f2fae2902cdedf47